src/components/view-pane/related-views-button.tsx

```diff
@@ -18,25 +22,38 @@ export class RelatedViewsButtonBase extends React.PureComponent<RelatedViewsButt
   }
 
   public render() {
+    const {config, relatedViews} = this.props;
+    const {hideRelatedViews} = config;
+    const hideCondition = (relatedViews.isHidden === undefined && hideRelatedViews) ||
```

Comment:
You still have redundant condition here, below for `onHideClick()`, and in the other file.   

Also, not sure why you have to write the condition in two different ways, although they should be the same. 

Since this is reused across file, why not consolidate the logic?

For example, you could just remove `relatedViews` and `config` from `RelatedViewsButtonProps`  and just have `hideRelatedViews` that you pass in (and `hideRelatedViews` should be already a combined version of what you have in relatedViews state and config state). 

Basically, related views button should be relatively dumb -- it doesn't have to know that externally you combine two properties for get the value whether you know related views or not.  

(See https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0, if you haven't read about it.)